### ISSUE TITLE
config mod `del` does not support multiple path matches

### ISSUE THREAD

**Issue by Grifs** [viash-io/viash#502]
This test fails:

```
  val baseJson: Json = parse(
    """foo:
      |  - name: bar
      |    a: 1
      |  - name: baz
      |    a: 1
      |  - name: qux
      |    a: 2
      |""".stripMargin).toOption.get
  
  test("test delete multiple entries from a list") {
    val expected1: Json = parse(
      """foo:
        |  - name: qux
        |    a: 2
        |""".stripMargin).toOption.get
    val cmd1 = ConfigModParser.block.parse("""del(.foo[.a == 1])""")
    val res1 = cmd1.apply(baseJson, false)
    assert(res1 == expected1)
  }
```

```
 * test delete multiple entries from a list - {
  "foo" : [
    {
      "name" : "baz",
      "a" : 1
    },
    {
      "name" : "qux",
      "a" : 2
    }
  ]
} did not equal {
  "foo" : [
    {
      "name" : "qux",
      "a" : 2
    }
  ]
}
```

To me it looks like it only ever deletes the first entry and then stops regardless of other possible path hits

**Comment by Grifs** [comment]
During testing a separate case emerged.
a config mod that deletes part of a matched path: `del(.foo[.a == 1].a)`

In the current version of the code this results in an equal faulty result:
```
"foo" : [
    {
      "name" : "bar"
    },
    {
      "name" : "baz",
      "a" : 1
    },
    {
      "name" : "qux",
      "a" : 2
    }
  ]
}
```
instead of 
```
"foo" : [
    {
      "name" : "bar"
    },
    {
      "name" : "baz",
    },
    {
      "name" : "qux",
      "a" : 2
    }
  ]
}
```

However, when re-enabling the code for the `Filter` case class in `PathExp.scala`
```
elemCursor = elemModified.top.get.hcursor.replay(elemCursor.history)
```
The above issue is fixed, however the original issue remains.

**Comment by Grifs** [comment]
FYI, these tests and some extra are added in the `fix/config_mod_del` branch

**Comment by Grifs** [comment]
Fixed in Viash 0.7.5
